Show controller without xr inputsource initial

Ledzz opened the pull request:
Add ability to show controller without xr inputsource

TODO:

- [x] Rename and move UnboundController from controller.tsx
- [ ] Add ability to pass gamepad state (maybe in next iteration)
- [x] Create separate demo for it
- [x] Update documentation

bbohlender commented on packages/react/xr/src/controller.tsx:
```diff
@@ -87,3 +101,72 @@ export const XRControllerModel = forwardRef<Object3D, XRControllerModelOptions>(
+  )
+}
+
+export function getXRControllerComponentObject(
```
I would move this function into /xr so its not only available for react users

bbohlender commented on packages/react/xr/src/controller.tsx:
```diff
@@ -87,3 +101,72 @@ export const XRControllerModel = forwardRef<Object3D, XRControllerModelOptions>(
+
+const unboundControllerContext = createContext<XRControllerState | undefined>(undefined)
+
+export const UnboundController: FC<{ profileIds: string[] }> = ({ profileIds, children }) => {
```
I wouldn't want to add a component for this use case since I dont think its that common
however, I'd appreciate a tutorial for this use case

Ledzz commented:
@bbohlender I've updated the PR: moved function to xr.tsx, created separate example with controller.
Then I plan to update documentation (add info about new hooks / functions, add tutorial).
Also I'd like to export XRControllerGamepadComponentId from the library, what do you think?

bbohlender commented on examples/react-three-xr/app.tsx:
```diff
@@ -1,5 +1,13 @@
+  XR,
+  XROrigin,
+  TeleportTarget,
+  UnboundController,
```
this seems incorrect?

bbohlender commented on packages/react/xr/src/xr.tsx:
```diff
@@ -138,3 +141,20 @@ export function useXR<T = XRState>(
+/**
+ * function for getting the object of a specific component from the xr controller model
+ */
+export function getXRControllerComponentObject(
```
I was rather saying that this function should be moved into packages/xr/ so vanilla users can also leverage the function ;-)

bbohlender reviewed (commented):
@Ledzz thanks for the changes, I can also adress these things, and then merge this PR asap ;-)

Ledzz commented:
@bbohlender updated with your comments and added some docs.
Could you please take a look? My example doesn't start now, I don't understand what I did wrong.
Could you please also recommend where to put docs for `getXRControllerComponentObject` function? Seems like it needs all-functions.md :)

bbohlender commented:
sure, I'll make final modifications and then merge if thats okay :)

Ledzz commented:
That's very kind of you, thank you! Let me know if I can help.
